code/modules/surgery/xeno.dm

```diff
@@ -0,0 +1,226 @@
+			SPAN_NOTICE("[user] starts to cut [target.caste_type] carapace. \the [tool], carefully, trying to prevent acid leaks."))
+	else
+		user.affected_message(target,
+			SPAN_NOTICE("You start to [pick("smash", "crack", "break")] [target.caste_type] carapace apart using \the [tool], Recklessly, with acid splashing on you!"),
+			SPAN_NOTICE("[user] starts to [pick("smash", "crack", "break")] Your carapace apart using \the [tool], Recklessly, with acid splashing all of the place!"),
+			SPAN_NOTICE("[user] starts to [pick("smash", "crack", "break")] [target.caste_type] carapace with \the [tool], Recklessly, with acid splashing them!"))
+		user.apply_damage(rand(25,50),BURN)
+		to_chat(user, SPAN_DANGER("You burn as acid gets on your skin!"))
+		//we dont really need log interact since we're working with dead body... I hope
+
+/datum/surgery_step/xenomorph/cut_exoskeleton/success(mob/living/carbon/human/user, mob/living/carbon/xenomorph/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	if(tool_type == /obj/item/tool/surgery/circular_saw)
+		user.affected_message(target,
+			SPAN_NOTICE("You succesfully cut through [target.caste_type] carapace apart using \the [tool]."),
+			SPAN_NOTICE("[user] Succesfully cuts through Your carapace using \the [tool]."),
+			SPAN_NOTICE("[user] Succesfully cuts [target.caste_type] carapace using \the [tool]."))
+	else
+		user.affected_message(target,
+			SPAN_NOTICE("You succesfully destroy [target.caste_type] carapace into bits and pieces apart using \the [tool]."),
+			SPAN_NOTICE("[user] succesfully destroys Your carapace into bits and pieces apart using \the [tool]."),,
+			SPAN_NOTICE("[user] Succesfully destroys [target.caste_type] carapace into bits and pieces apart using \the [tool]."))
+	for(var/mob/living/carbon/human/victim in orange(1, target))
+		if(istype(victim.wear_suit, /obj/item/clothing/suit/bio_suit) && istype(victim.head, /obj/item/clothing/head/bio_hood))
+			continue
+		to_chat(victim, SPAN_HIGHDANGER("You are covered in acid as you feel agonizing pain!"))
+		victim.apply_damage(rand(75, 125), BURN) // you WILL wear biosuit.
+		playsound(victim, "acid_sizzle", 25, TRUE)
+		animation_flash_color(victim, "#FF0000")
+
+/datum/surgery_step/xenomorph/cut_exoskeleton/failure(mob/living/carbon/human/user, mob/living/carbon/xenomorph/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	if(tool_type == /obj/item/tool/surgery/circular_saw)
+		user.affected_message(target,
+			SPAN_WARNING("Your hand slips, failing to cut [target.caste_type] carapace apart using \the [tool]!"),
+			SPAN_WARNING("[user] hand slips, failing to cut Your carapace apart using \the [tool]!"),
+			SPAN_WARNING("[user] hand slips, failing to cut [target.caste_type] carapace using \the [tool]!"))
+	else
+		user.affected_message(target,
+			SPAN_WARNING("Your hand slips, failing to destroy [target.caste_type] carapace into bits and pieces apart using \the [tool]."),
```

Comment:
the apart here is redundant, you already said it's not been destroyed into bits and pieces

Reply:
where do I do that??